smallfile-wrapper/smallfile-wrapper.py

```diff
@@ -0,0 +1,119 @@
+        item["user"] = user
+        processed.append(item)
+    return processed
+
+
+def main():
+
+    # retrieving elasticsearch server address and operating port from
+    # environement variables. The credentials are exported in the
+    # tasks of ripsaw/roles/smallfile-bench. These variables are exported
+    # based on condition, if the es_port and es_server is defined in
+    # smallfile CRDS of ripsaw. Use of "os.environ.get($env_var)" is to look
+    # for the supplied $env_var. If there is no such $env_var defined,
+    # then it will return None object.
+    server = os.environ.get("es")
+    # Use of "os.environ.get($env_var)" is to look
+    # for the supplied $env_var. If there is no such $env_var defined,
+    # then it will return None object.
+    port = os.environ.get("es_port")
+    # for each benchmark run, ripsaw generates a UUID and export it
+    # as a environement variable. Here, that uuid and a test_user is
+    # is trying to be fetched from env. Use of "os.environ.get($env_var)"
+    # is to look for the supplied $env_var. If there is no such $env_var
+    # defined, then it will return None object.
+    uuid = os.environ.get("uuid")
+    user = os.environ.get("test_user")
+
+    # Smallfile results path is based on the run, if it is executed on
+    # pv or without PV. Hence, {{smallfile_path}} is supplied at stdin
+    # to determine the RESULTS directory
+    smallfile_path = sys.argv[1]
+    # The number of result JSON files depends on the number of operations
+    # executed by smallfile. Sequence of operations is supplied through
+    # smallfile CRD in ripsaw
+    operations = sys.argv[2:]
+
+    # Consolidating all the JSONs in to 1 list here. "data" is
+    # final list which has all the smallfile results. The format is
+    # [{smallfile_result_1},{smallfile_result_2},{smallfile_result_3}]
+    data = _load_result_JSON(smallfile_path, operations)
+    # Now, smallfile results are available in python memory, before
+    # passing this data to Elasticsearch, it has to be injected
+    # with the $uuid and $user, so as to make the result, easily
+    # and uniquely identifiable on Elasticsearch. If $uuid and $user
+    # is not fetched from env, the script will end here.
+    if (uuid is not None) and (user is not None):
+        documents = _json_payload(data, uuid, user)
+        # By reaching here, means the $uuid and $user has been successfully
+        # retrieved and injected in each piece of smallfile result. Now, data
+        # is all set to send to elasticsearch server. Checking for ES
+        # server and port address here. If available, _index_result() will s
+        # establish connection to ES server and send data, by iterating through
+        # each smallfile result in the "documents" list.
+        if (server is not None) and (port is not None):
+            if len(documents) > 0:
+                status = _index_result(server, port, documents)
+                if status:
```

Comment:
_index_result only returns false if a failure happens, it never returns the status...

Reply:
Thanks for pointing this out, adding a return statement , when its done.